manifests/osfamily/solaris.pp

```diff
@@ -32,6 +32,57 @@
         source => "puppet:///pe_packages/${pe_server_version}/${::platform_tag}/solaris-noask",
       }
     }
+    '11': {
+      class { 'puppet_agent::prepare::package':
+        package_file_name => $package_file_name,
+      }
+      contain puppet_agent::prepare::package
+
+      # Backup user configuration because solaris 11 will blow away
+      # /etc/puppetlabs/ when uninstalling the pe-* modules.
+      file { '/tmp/puppet_agent/':
+        ensure => directory,
+      }
+      exec { 'puppet_agent backup /etc/puppetlabs/':
+        command => 'cp -r /etc/puppetlabs/ /tmp/puppet_agent/',
+        require => [Exec['puppet_agent copy packages'], File['/tmp/puppet_agent/']],
+        path    => '/bin:/usr/bin:/sbin:/usr/sbin',
+      }
+
+      $pkgrepo_dir = '/etc/puppetlabs/installer/solaris.repo'
+
+      exec { 'puppet_agent remove existing repo':
+        command   => "pkgrepo remove -s '${pkgrepo_dir}' '*'",
```

Comment:
This fails on my solaris 11.2 & 11.3 i386 nodes.  The `pkgrepo remove` command removes packages from within the repository, not the repository itself.  The repo is already empty (part of install process I believe) and the remove command results in an  exit code of 1.  I fixed by changing the command into:

`command   => "rm -rf '${pkgrepo_dir}'/'*'",`

The `pkgrepo create` that follows creates a new empty repository.  After making this change I successfully tested it on my solaris 11.2 i386 node.